weatherApp/R/weather_app_functions.R

```diff
@@ -0,0 +1,371 @@
+### weather_app in R Shiny ###
+
+library(httr)
+library(jsonlite)
+library(dplyr)
+library(leaflet)
+library(mapview)
+library(magick)
+library(stringr)
+# library(tidygeocoder)
+
+# you will need an API key to access the data from openweathermaps.org
+# you have to sign up for a free account to get the api key
+# the api key is private and should not be shared with other people
+
+# my_api_key <- "insert your api key here"
+
+# tidygeocoder::geo_osm() uses a string with an address as input and returns
+# the latitude and longitude of the location. The tidygeocoder::geo_osm() function was substituted with my own function geocode().
+
+# The function geocode() takes a location as input and returns the longitude and latitude of this location.
+# This way, I don't need to install and use the external function tidygeocoder::geo_osm() anymore.
+
+#' @export
+geocode <- function(location){
+
+  # split the location string, that the function got as input, into the different words.
+  my_location <- stringr::str_split(location, boundary("word"))
+  my_location_string <- my_location[[1]][1]
+
+  for(i in 2:length(my_location[[1]])){
+    my_location_string <- paste(my_location_string, "+", my_location[[1]][i], sep = "")
+  }
+
+  # Write down the url that retrieves the long/lat information from openstreetmap.org
+
+  my_url <- paste("https://nominatim.openstreetmap.org/search?q=", my_location_string,
+                  "&format=json&limit=1", sep="")
+
+  # call the Url, retrieve the data and check if it was successful.
+  my_raw_results <- httr::GET(my_url)
+
+  if (status_code(my_raw_results)[[1]] != 200) {
+    stop("Please enter a valid location!")
+  }
+
+  # Transform the JSON data into useable format.
+  my_content <- httr::content(my_raw_results, as = "text")
+
+  my_content_from_json <- jsonlite::fromJSON(my_content)
+
+  # Save the longitude latitude data in an object and return it.
+  geocode_data <- data.frame(3)
+  geocode_data$location <- location
+  geocode_data$latitude <- my_content_from_json$lat
+  geocode_data$longitude <- my_content_from_json$lon
+
+  return(geocode_data)
+
+}
+
+# Example for geocode()
```

Comment:
Eventually, you could quickly reiterate what to expect from the example code since I at least already forgot the functionality after reading until the examples.